core/src/main/java/io/grpc/inprocess/InProcessServerBuilder.java

```diff
@@ -69,6 +74,38 @@ public static InProcessServerBuilder forName(String name) {
     return new InProcessServerBuilder(name);
   }
 
+  /**
+   * Builds an anonymous (e.g. unregistered) in-process server that exposes a given underlying
+   * server. The returned in-process server will have already been started.
+   *
+   * @param server the server that the in-process server is exposing
+   * @return an in-process form of the given server
+   */
+  static InProcessServer anonymous(ServerImpl server) {
+    InProcessServerBuilder builder = new InProcessServerBuilder(server.handlerRegistry());
+    Executor executor = server.executor();
+    if (executor != SharedResourceHolder.get(GrpcUtil.SHARED_CHANNEL_EXECUTOR)) {
```

Comment:
Calling get() increases a reference count and may create it. I'll try to come up with something less likely to cause problems.